src/utils/rfq_blockchain_utils.ts

```diff
@@ -129,6 +156,61 @@ export class RfqBlockchainUtils {
         return this._exchangeProxy.executeMetaTransaction(metaTx, metaTxSig).getABIEncodedTransactionData();
     }
 
+    public async getNonceAsync(workerAddress: string): Promise<number> {
+        return this._web3Wrapper.getAccountNonceAsync(workerAddress);
+    }
+
+    public getExchangeProxyAddress(): string {
+        return this._exchangeProxyAddress;
+    }
+
+    public async getTransactionReceiptIfExistsAsync(transactionHash: string): Promise<TransactionReceipt | undefined> {
+        return this._web3Wrapper.getTransactionReceiptIfExistsAsync(transactionHash);
+    }
+
+    public async getCurrentBlockAsync(): Promise<number> {
+        return this._web3Wrapper.getBlockNumberAsync();
+    }
+
+    public async estimateGasForExchangeProxyCallAsync(callData: string, workerAddress: string): Promise<number> {
+        const txData: Partial<TxData> = {
+            to: this._exchangeProxy.address,
+            data: callData,
+            from: workerAddress,
+        };
+
+        const gasEstimate = await this._web3Wrapper.estimateGasAsync(txData);
+
+        // add a buffer
+        return Math.ceil((GAS_ESTIMATE_BUFFER + 1) * gasEstimate);
+    }
+
+    public getRfqOrderTakerTokenFilledAmountFromLogs(logs: LogEntry[]): BigNumber {
```

Comment:
Can this method be written in terms of `getDecodedRfqOrderFillEventLogFromLogs`?

Reply:
it definitely can be, I'm actually just going to get rid of it. It's not necessary now